### PR TITLE
Add commit funds

### PR DESCRIPTION
**Pull request by banciur**

### CODE REVIEW

File: app/reducers/index.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { combineReducers } from "redux";
+import { reducer as form } from "redux-form";
 import icoState, { IcoState } from "./icoState";
 
 export interface IAppState {
   readonly icoState: IcoState;
+  readonly form: any;
```
Comment (krzkaczor) [inline app/reducers/index.ts]: You can get state shape from `redux-form` library - it's `FormState`

File: app/muiTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import getMuiTheme from "material-ui/styles/getMuiTheme";
+// colors should be same as in common/sass/variables.scss
+export default getMuiTheme({
+  palette: {
+    primary1Color: "#d5e20f",
```
Comment (krzkaczor) [inline app/muiTheme.tsx]: this is totally fine but I found webpack loader that can do load sass variables :D https://github.com/nordnet/sass-variable-loader but let's dont over compilcate things for now

File: app/index.tsx [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 import reduxLogger from "redux-logger";
 import reduxThunk from "redux-thunk";
 import App from "./containers/App";
 import { Commit } from "./containers/Commit";
+import muiTheme from "./muiTheme";
 import reducers from "./reducers";
 
 // @todo add bundle splitting and separate these renders #UAF
+
+injectTapEventPlugin();
```
Comment (krzkaczor) [inline app/index.tsx]: tbh i never used it before. Do we need it still today? https://www.reddit.com/r/reactjs/comments/4pe61k/why_do_we_need_reacttapeventplugin_in_our_projects/ i think material-ui doesnt rely on it anymore

File: app/containers/Commit.tsx [hunk 4]
```diff
@@ -1,8 +1,56 @@
...
 export const Commit: React.SFC = () => {
-  return <CommitNavbar>Commit funds in Neufund Commitment Opportunity</CommitNavbar>;
+  return (
+    <div className={style.pageContainer}>
+      <CommitNavbar>Commit funds in Neufund Commitment Opportunity</CommitNavbar>
+      <Row>
+        <Col sm={10} smOffset={1} className={style.contentContainer}>
+          <Row>
+            <Col sm={12}>
+              <h2 className={style.sectionHeader}>
```
Comment (krzkaczor) [inline app/containers/Commit.tsx]: what about extracting it to separe component? Like <TitleHeader number={1}>Commit funds</TitleHeader>

File: app/containers/Commit.tsx [hunk 5]
```diff
@@ -1,8 +1,56 @@
...
+import { Col, Row } from "react-bootstrap";
+import { CommitFundsDescription } from "../components/CommitFundsDescription";
+import CommitFundsForm from "../components/CommitFundsForm";
 import { CommitNavbar } from "../components/CommitNavbar";
+import { UserAddressComponent } from "../components/UserAddressComponent";
+import * as style from "./Commit.scss";
 
 export const Commit: React.SFC = () => {
-  return <CommitNavbar>Commit funds in Neufund Commitment Opportunity</CommitNavbar>;
+  return (
```
Comment (krzkaczor) [inline app/containers/Commit.tsx]: please extract whole body to separate dumb component taking all hardcoded values like 0x6ddfa40a2631348c2bd4b0c949ade1712b44587641c3309c5e8d2b914151ed50 as props. And in this component just use the newly created one pushing all hardcoded values. This way we can easily connect redux to this component. You can name newly created component as "CommitComponent" and keep it in this file. default export will be exporting connected (hardcoded for now) component

File: app/components/UserAddressComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+interface IUserAddressComponentProps {
+  address: string;
+}
+
+export const UserAddressComponent: React.SFC<IUserAddressComponentProps> = ({ address }) =>
+  <div className={userAddressContainer}>
+    <AddressIcon address={address} className={icon} />
+    <p>Your wallet address</p>
+    <p className={addressStyle}>
+      <TextCopyable text={address} maxTextLength={8} />
```
Comment (krzkaczor) [inline app/components/UserAddressComponent.tsx]: What about using https://www.npmjs.com/package/react-copy-to-clipboard for copy instead?

File: app/containers/Commit.tsx (hunk 5 above)
Comment (banciur) [reply app/containers/Commit.tsx]: fixed

File: app/containers/Commit.tsx (hunk 4 above)
Comment (banciur) [reply app/containers/Commit.tsx]: fixed

File: app/reducers/index.ts (hunk 1 above)
Comment (banciur) [reply app/reducers/index.ts]: done

File: app/index.tsx (hunk 3 above)
Comment (banciur) [reply app/index.tsx]: fixed

**Pull request merged**